src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -34,6 +63,16 @@ public override object Get(DbDataReader rs, string name, ISessionImplementor ses
 			return Convert.ToString(rs[name]);
 		}
 
+		public override bool IsEqual(object x, object y)
+		{
+			return (Comparer ?? DefaultComparer)?.Equals(x, y) ?? base.IsEqual(x, y);
```

Comment:
`null` comparer is the same as `Ordinal`, so it does not make sense to piling up null propagation and condition operators. It should be simplified to just `return Comparer.Equals(x, y)` (see notes on `Comparer` and `DefaultComparer` properties for context).